development/idp-local/UserPass.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace SimpleSAML\Module\exampleauth\Auth\Source;
+
+/**
+ * Example authentication source - username & password.
+ *
+ * This class is an example authentication source which stores all username/passwords in an array,
+ * and authenticates users against this array.
+ *
+ * @author Olav Morken, UNINETT AS.
+ * @package SimpleSAMLphp
+ */
+
+class UserPass extends \SimpleSAML\Module\core\Auth\UserPassBase
+{
+    /**
+     * Our users, stored in an associative array. The key of the array is "<username>:<password>",
+     * while the value of each element is a new array with the attributes for each user.
+     */
+    private $users;
+
+    /**
+     * Constructor for this authentication source.
+     *
+     * @param array $info  Information about this authentication source.
+     * @param array $config  Configuration.
+     */
+    public function __construct($info, $config)
+    {
+        assert(is_array($info));
+        assert(is_array($config));
```

Comment:
I didn't want to change anything that wasn't necessary for the process of merging the module into this repo. One of the necessary changes for SSP 2.x is to use type hinting, so that's going to happen soon.